WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
```

Comment:
There's no advantage in sticking to using `self.managedObjectContext` in this class (even if it's used similarly in other classes), and in fact it can potentially cause an exception here.

I realize this pattern may be used elsewhere but that doesn't mean it's a good one.

No worries though, I'll think about it a bit more and will eventually come up with a separate ticket for this.